bug: sent_tokenize with engine="whitespace" should return [] not [[]]

### Description

`sent_tokenize(str, engine="whitespace")` returns a list of a list of string, instead of a list of string.

### Expected results

`sent_tokenize` should return a list (`[]`), not a list of a list (`[[]]`)

### Current results

https://github.com/PyThaiNLP/pythainlp/actions/runs/11627805664/job/32381817268


```
======================================================================
FAIL: test_sent_tokenize (tests.test_tokenize.TokenizeTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/runner/work/pythainlp/pythainlp/tests/test_tokenize.py", line 222, in test_sent_tokenize
    self.assertEqual(
AssertionError: Lists differ: [['รักน้ำ', 'รักปลา', '']] != ['รักน้ำ', 'รักปลา', '']

First differing element 0:
['รักน้ำ', 'รักปลา', '']
'รักน้ำ'

Second list contains 2 additional elements.
First extra element 1:
'รักปลา'

- [['รักน้ำ', 'รักปลา', '']]
? -                        -

+ ['รักน้ำ', 'รักปลา', '']
```

### Steps to reproduce

```python
        self.assertEqual(
            sent_tokenize("รักน้ำ  รักปลา  ", engine="whitespace"),
            ["รักน้ำ", "รักปลา", ""],
        )
```

### PyThaiNLP version

5.0.4

### Python version

All

### Operating system and version

All

### More info

_No response_

### Possible solution

_No response_

### Files

_No response_

--- a/pythainlp/tokenize/core.py
+++ b/pythainlp/tokenize/core.py
@@ -538,7 +538,7 @@ def sent_tokenize(
         result = map_indices_to_words(word_indices, segments)
         return result
     else:
-        return [segments]
+        return segments
 
 
 def paragraph_tokenize(

--- a/tests/test_tokenize.py
+++ b/tests/test_tokenize.py
@@ -225,18 +225,6 @@ class TokenizeTestCase(unittest.TestCase):
             sent_tokenize("รักน้ำ  รักปลา  ", engine="whitespace+newline"),
             ["รักน้ำ", "รักปลา"],
         )
-        self.assertEqual(
-            sent_tokenize(SENT_1),
-            SENT_1_TOKS,
-        )
-        self.assertEqual(
-            sent_tokenize(SENT_2),
-            SENT_2_TOKS,
-        )
-        self.assertEqual(
-            sent_tokenize(SENT_3),
-            SENT_3_TOKS,
-        )
         self.assertIsNotNone(
             sent_tokenize(
                 SENT_1,

--- a/tests/testx_tokenize.py
+++ b/tests/testx_tokenize.py
@@ -92,6 +92,18 @@ class TokenizeTestCaseX(unittest.TestCase):
         # Use default engine (crfcut)
         self.assertEqual(sent_tokenize(None), [])
         self.assertEqual(sent_tokenize(""), [])
+        self.assertEqual(
+            sent_tokenize(SENT_1),
+            SENT_1_TOKS,
+        )
+        self.assertEqual(
+            sent_tokenize(SENT_2),
+            SENT_2_TOKS,
+        )
+        self.assertEqual(
+            sent_tokenize(SENT_3),
+            SENT_3_TOKS,
+        )
 
         self.assertEqual(
             sent_tokenize(SENT_1, engine="crfcut"),
@@ -106,16 +118,8 @@ class TokenizeTestCaseX(unittest.TestCase):
             SENT_3_TOKS,
         )
         self.assertEqual(
-            sent_tokenize(SENT_1),
-            SENT_1_TOKS,
-        )
-        self.assertEqual(
-            sent_tokenize(SENT_2),
-            SENT_2_TOKS,
-        )
-        self.assertEqual(
-            sent_tokenize(SENT_3),
-            SENT_3_TOKS,
+            sent_tokenize(SENT_4, engine="crfcut"),
+            [["ผม", "กิน", "ข้าว", " ", "\n", "เธอ", "เล่น", "เกม"]],
         )
         self.assertIsNotNone(
             sent_tokenize(
@@ -135,6 +139,7 @@ class TokenizeTestCaseX(unittest.TestCase):
                 engine="tltk",
             ),
         )
+
         self.assertIsNotNone(
             sent_tokenize(
                 SENT_1,
@@ -153,6 +158,11 @@ class TokenizeTestCaseX(unittest.TestCase):
                 engine="thaisum",
             ),
         )
+        self.assertEqual(
+            sent_tokenize(SENT_4, engine="thaisum"),
+            [["ผม", "กิน", "ข้าว", " ", "เธอ", "เล่น", "เกม"]],
+        )
+
         self.assertIsNotNone(
             sent_tokenize(
                 SENT_3,
@@ -177,14 +187,6 @@ class TokenizeTestCaseX(unittest.TestCase):
         #         engine="wtp-large",
         #     ),
         # )
-        self.assertEqual(
-            sent_tokenize(SENT_4, engine="crfcut"),
-            [["ผม", "กิน", "ข้าว", " ", "\n", "เธอ", "เล่น", "เกม"]],
-        )
-        self.assertEqual(
-            sent_tokenize(SENT_4, engine="thaisum"),
-            [["ผม", "กิน", "ข้าว", " ", "เธอ", "เล่น", "เกม"]],
-        )
 
     def testx_word_tokenize(self):
         self.assertIsNotNone(word_tokenize(TEXT_1, engine="nlpo3"))

